src/mono/msbuild/AppleAppBuilder/AppleAppBuilder.cs

```diff
@@ -203,4 +178,57 @@ public override bool Execute()
 
         return true;
     }
+
+    static void GenerateLinkAllFile(IEnumerable<string> asmFiles, string outputFile)
```

Comment:
So the AOTCompilation task won't be responsible for this?

Reply:
nit: explicit access modifier

Reply:
hmm good question, it sounds pretty iOS specific to me so I don't think it makes sense in the shared AOT task.

Reply:
I'd say it's FullAOT-specific

Reply:
It's specifically because of the `--aot=static` option that iOS is using and from what I can see XA doesn't right now. 

We'd have to make it generic enough to be useful (e.g. deal with the `TARGET_OS_IPHONE` defines, `<TargetConditionals.h>` includes, .m vs. .c etc.) so I'd leave it separate for now.